tests/test_buffers.py

```diff
@@ -140,13 +140,16 @@ def test_device_buffer(replay_buffer_cls, device):
     if replay_buffer_cls in [RolloutBuffer, DictRolloutBuffer]:
         data = buffer.get(50)
     elif replay_buffer_cls in [ReplayBuffer, DictReplayBuffer]:
-        data = buffer.sample(50)
+        data = [buffer.sample(50)]
 
     # Check that all data are on the desired device
     desired_device = get_device(device).type
-    for value in list(data):
-        if isinstance(value, dict):
-            for key in value.keys():
-                assert value[key].device.type == desired_device
-        elif isinstance(value, th.Tensor):
-            assert value.device.type == desired_device
+    for minibatch in list(data):
```

Comment:
There's only one pass in this loop, isn't there? If so, I think you can simply delete it and drop the brackets from line 143.

Reply:
There are two passes actually, because `.get(50)` returns a Generator for two batches of size 50. There are 2 batches because the buffer is of size 100.

Reply:
The reason line 143 needs the brackets is to make sure both branches (`.get()` and `[.sample()]` ) are iterables with the same type

Reply:
makes sens :)